### ISSUE TITLE
[$250] [Due for payment 2025-03-31] Add VoiceOver for chat list items

### ISSUE THREAD

**Issue by dubielzyk-expensify** [Expensify/App#56993]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Turn on VoiceOver in (Settings -> Accessibility -> VoiceOver)
2. Go to the Expensify App
3. Go to the Inbox tab/screen
4. Select a chat item in the chat list
5. Listen to the VoiceOver label

## Expected Result:
Expected the VoiceOver to read the contents of a chat list

## Actual Result:
VoiceOver reads "Navigates to a chat. Button" instead which isn't helpful as you can't differentiate the items in the list

## Workaround:
Don't think there is one.

## Area issue was found in:
iOS and macOS devices

## Failed WCAG checkpoints
What WCAG checkpoints were failed?
WCAG 4.1 as it's not compatible with assistive technologies. Blind or any users relying on Apples VoiceOver or other screen reading technologies would fall under this issue.

## User impact:
How does this affect the user?
VoiceOver users would largely struggle to use 30% of the navigation of the app and a key highlight feature

## Suggested resolution:
Add aria labels or similar treatment that's seen on the `Reports` screen

## Platforms:

Which of our officially supported platforms is this issue occurring on? Please only tick the box if you have provided a screen-recording in the thread for each platform:
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

**Version Number:** v9.0.95-1
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** N/A
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/user-attachments/assets/0dca7dfa-0bf1-4c75-b00c-d6fc27844604

**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021906782372327213826</li>
        <li>Upwork Job ID: 1906782372327213826</li>
        <li>Last Price Increase: 2025-03-31</li>
<li>Automatic offers: </li>
<ul>
<li>DylanDylann | Reviewer | 106735495</li>
<li>thelullabyy | Contributor | 106735498</li>
</ul></li>
    </ul>
</details>

**Comment by dubielzyk-expensify** [comment]
cc @Expensify/design for visibility

**Comment by thelullabyy** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
VoiceOver reads "Navigates to a chat. Button" instead which isn't helpful as you can't differentiate the items in the list

### What is the root cause of that problem?
The VoiceOver based on:
https://github.com/Expensify/App/blob/bf23769f7ee8e1dfabfcb803a8a1dde54413b670/src/components/LHNOptionsList/OptionRowLHN.tsx#L238
and now, it is only _"Navigates to a chat"_
### What changes do you think we should make in order to solve the problem?
We can update to ```accessibilityLabel={`${translate('accessibilityHints.navigatesToChat')} ${optionItem.text}`}```. The exact text to be shown can be discussed later.

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?
In SidebarTest.ts, verify that the accessibility label for each option row is rendered properly

### What alternative solutions did you explore? (Optional)

**Comment by DylanDylann** [comment]
@daledah's [proposal](https://github.com/Expensify/App/issues/56993#issuecomment-2670528072) looks good to me 

Note for the PR phase: we need to discuss the correct message with the design team

This is the current display after applying the above proposal

https://github.com/user-attachments/assets/b61a6d10-20eb-47ef-b240-4ab422c845c1


🎀 👀 🎀 C+ Reviewed

**Comment by NikkiWines** [comment]
Agreed, @thelullabyy's proposal looks good 👍

**Comment by dubielzyk-expensify** [comment]
If we wanna be consistent with the iOS Apple messages I think we should say something like:
`[Name of chat]. ["Unread" if unread chat]. [Preview message]`. 

Examples:
`Jon Dubielzyk. Unread. Yeah I can send this to you`
`Shawn Borton. Sending you an expense now`


I could see us say something like `Go to chat.` At the end as well. Usually VoiceOver mentions the available actions you can do which we don't seem to do. We seem to just say `button` but that seems weird to me in this instance.

**Comment by garrettmknight** [comment]
@thelullabyy when can we expect your PR?

**Comment by garrettmknight** [comment]
@DylanDylann can you complete the checklist when you get a chance?

**Comment by garrettmknight** [comment]
@thelullabyy / @DylanDylann please accept the offers and I'll pay out once the checklist is done.

**Comment by DylanDylann** [comment]
# BugZero Checklist:

- [ ] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [ ] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment:

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [ ] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

<details>
<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1.

Do we agree 👍 or 👎


</details>

**Comment by garrettmknight** [comment]
Payment Summary:
- contributor: @thelullabyy $250 paid via upwork
- contributor: @DylanDylann $250 paid via upwork
